crypto/ml_kem/ml_kem.c

```diff
@@ -12,89 +23,119 @@
 int ml_kem_512_ipd_keypair_deterministic(uint8_t *public_key  /* OUT */,
                                          uint8_t *secret_key  /* OUT */,
                                          const uint8_t *seed  /* IN */) {
-  return ml_kem_512_ref_keypair_derand(public_key, secret_key, seed);
+  ml_kem_params params;
+  ml_kem_512_params_init(&params);
```

Comment:
Functions are specific to a parameter set, so don't need a function that construct the parameter on every invocation, You can simply define a static object for the 3 parameter sets and use that? No need to construct them for every invocation.

Reply:
Note that if you carry this into the fips module you would need to use the BSS allocation machinery.

Reply:
yeah, I decided to use a function instead of static object to avoid having to do the BSS machinery when we move the code to the FIPS module.